scripting/get5/recording.sp

```diff
@@ -94,9 +94,63 @@ public Action Timer_FireStopRecordingEvent(Handle timer, DataPack pack) {
   Call_PushCell(event);
   Call_Finish();
   EventLogger_LogAndDeleteEvent(event);
+
+  UploadDemoToServer(demoFileName);
   return Plugin_Handled;
 }
 
+void UploadDemoToServer(const char[] demoFileName) {
+  bool steamWorksAvaliable = LibraryExists("SteamWorks");
+
+  if (!steamWorksAvaliable) {
+    LogDebug("Cannot upload demoes to a web server without the SteamWorks extension running.");
+    return;
+  }
+
+  char demoUrl[1024];
+  char demoAuthKey[1024];
+  char demoAuthValue[1024];
+
+  g_DemoUploadUrlCvar.GetString(demoUrl, sizeof(demoUrl));
+  g_DemoUploadAuthKeyCvar.GetString(demoAuthKey, sizeof(demoAuthKey));
+  g_DemoUploadAuthValueCvar.GetString(demoAuthValue, sizeof(demoAuthValue));
+
+  if (StrEqual(demoUrl, "")) {
+    LogDebug("Will not upload any demoes as upload URL is not set.");
+    return;
+  }
+
+  char formattedUrl[1024];
+  strcopy(formattedUrl, sizeof(formattedUrl), demoUrl);
+  ReplaceString(formattedUrl, sizeof(formattedUrl), "\"", "");
+
+  // Add the protocol strings if missing (only http).
+  if (StrContains(formattedUrl, "http://") == -1 && StrContains(formattedUrl, "https://") == -1) {
+    Format(formattedUrl, sizeof(formattedUrl), "http://%s", formattedUrl);
+  }
+
+  LogDebug("UploadDemoToServer: formattedUrl (SteamWorks) = %s", formattedUrl);
+  Handle demoRequest = SteamWorks_CreateHTTPRequest(k_EHTTPMethodPOST, formattedUrl);
+  // Store the filename of the demo for the callback.
+  Format(g_DemoFileNameLastUploaded, sizeof(g_DemoFileNameLastUploaded), demoFileName);
```

Comment:
Move this down to `SteamWorks_SendHTTPRequest` inside an if, so it doesn't get set if the request is not sent - you could also add a LogError there if you want.

```c
if (SteamWorks_SendHTTPRequest(demoRequest)) {
  Format(g_DemoFileNameLastUploaded, sizeof(g_DemoFileNameLastUploaded), demoFileName);
}
```